x-pack/solutions/security/plugins/security_solution/common/trial_companion/constants.ts

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import type { NBA, NBAAction } from './types';
+import { Milestone } from './types';
+
+export const TRIAL_COMPANION_NBA_URL = '/internal/security_solution/trial_companion/nba';
+export const TRIAL_COMPANION_NBA_ACTION_URL =
+  '/internal/security_solution/trial_companion/nba/action';
+
+const SUFFIX_MESSAGE = 'message';
+const SUFFIX_TITLE = 'title';
+const SUFFIX_ACTION = 'action';
+
+function translateNBA(milestone: Milestone, message: string, suffix: string) {
+  return i18n.translate(`xpack.securitySolution.trialCompanion.nba.m${milestone}.${suffix}`, {
+    defaultMessage: message,
+  });
+}
+
+function toNBA(milestoneId: Milestone, message: string, title: string, actions?: NBAAction[]): NBA {
+  const apps: NBAAction[] | undefined = actions?.map((a) => {
+    return {
+      app: a.app,
+      text: translateNBA(milestoneId, a.text, SUFFIX_ACTION),
+    } as NBAAction;
+  });
+  return {
+    message: translateNBA(milestoneId, message, SUFFIX_MESSAGE),
+    title: translateNBA(milestoneId, title, SUFFIX_TITLE),
+    apps,
+  } as NBA;
+}
+
+const NBA_M1: NBA = toNBA(
```

Comment:
Some comments

Why not just define a class implementing the `NBA` interface instead of declaring helper/builder methods to create NBA instances? Or as an alternative, implement some sort of builder pattern, like, `NBA.builder().id(..).msg(...).action(...).action(...).build()`
Also, since the file name is `constants.ts`, it might not be the best place to declare functions.
Finally, I think the i18n logic should not be coupled to the NBA itself since it's just for UI purposes. Why not just keep the MBA model simple and translate the title or whatever should be translated when it's required?